util/grpc/sanitizer.go

```diff
@@ -0,0 +1,73 @@
+package grpc
+
+import (
+	"errors"
+	"strings"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	contextKey = "sanitizer"
+)
+
+// ErrorSanitizerUnaryServerInterceptor returns a new unary server interceptor that sanitizes error messages
+// and provides Sanitizer to define replacements.
+func ErrorSanitizerUnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		sanitizer := NewSanitizer()
+		resp, err = handler(ContextWithSanitizer(ctx, sanitizer), req)
+		if err == nil {
+			return resp, nil
+		}
+
+		if se, ok := err.(interface{ GRPCStatus() *status.Status }); ok {
+			return resp, status.Error(se.GRPCStatus().Code(), sanitizer.Replace(se.GRPCStatus().Message()))
+		}
+
+		return resp, errors.New(sanitizer.Replace(err.Error()))
+	}
+}
+
+// ContextWithSanitizer returns a new context with sanitizer set.
+func ContextWithSanitizer(ctx context.Context, sanitizer Sanitizer) context.Context {
+	return context.WithValue(ctx, contextKey, sanitizer)
+}
+
+// SanitizerFromContext returns sanitizer from context.
+func SanitizerFromContext(ctx context.Context) (Sanitizer, bool) {
+	res, ok := ctx.Value(contextKey).(Sanitizer)
+	return res, ok
+}
+
+// Sanitizer provides methods to define list of strings and replacements
+type Sanitizer interface {
+	Replace(s string) string
+	AddReplacement(val string, replacement string)
+}
+
+type sanitizer struct {
+	replacements map[string]string
```

Comment:
I feel the maintenance of a replacements map[string]string is more maintenance and more granular than we need. Could we instead give the sanitizer a root path (e.g. `/tmp/argocd_repos`) and then the replacement process would replace everything one subdirectory deeper with `.` just using regular expression replacement?

This way there is no mapping to maintain since we by convention, understand that every repo is cloned as a sub directory to the repo root directory.

Reply:
answered in https://github.com/argoproj/argo-cd/pull/8517#discussion_r812390299